File: frontend/src/types/cronexecutionlog.ts
/**
 * 执行日志相关类型定义
 */

/**
 * 日志级别
 */
export type LogLevel = 'info' | 'warn' | 'error' | 'debug';

/**
 * 执行日志详情
 */
export interface CronExecutionLog {
  id: number;
  execution_id: number;
  from: string;
  level: LogLevel;
  message: string;
  created_at: string;
  updated_at: string;
}

/**
 * 执行日志列表查询参数
 */
export interface ExecutionLogListParams {
  page?: number;
  pageSize?: number;
  execution_id?: number;
  level?: LogLevel;
  from?: string;
}

File: backend/internal/model/cron_execution_log.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// CronExecutionLog 定时任务执行日志
type CronExecutionLog struct {
	ID uint `gorm:"primarykey"`

	ExecutionID uint   `gorm:"not null;index;comment:'关联执行记录 ID'"`
	From        string `gorm:"type:varchar(100);not null;comment:'日志来源阶段'"`
	Level       string `gorm:"type:varchar(20);not null;comment:'日志级别 (info/warn/error/debug)'"`
	Message     string `gorm:"type:text;not null;comment:'日志内容'"`

	CreatedAt time.Time      `gorm:"index:,sort:desc;"`
	UpdatedAt time.Time
	DeletedAt gorm.DeletedAt `gorm:"index"`
}

func (CronExecutionLog) TableName() string {
	return "cron_execution_log"
}
